wandb/sdk/internal/job_builder.py

```diff
@@ -154,32 +165,36 @@ def _build_repo_job(
 
         else:
             full_program_path = program_relpath
+
         # TODO: update executable to a method that supports pex
         source: GitSourceDict = {
+            "git": {
+                "remote": remote,
+                "commit": commit,
+            },
             "entrypoint": [
                 os.path.basename(sys.executable),
                 full_program_path,
             ],
             "notebook": self._is_notebook_run(),
-            "git": {
-                "remote": remote,
-                "commit": commit,
-            },
         }
-
         name = make_artifact_name_safe(f"job-{remote}_{program_relpath}")
 
-        artifact = JobArtifact(name)
-        if os.path.exists(os.path.join(self._settings.files_dir, DIFF_FNAME)):
-            artifact.add_file(
-                os.path.join(self._settings.files_dir, DIFF_FNAME),
-                name=DIFF_FNAME,
+        # if job is a partial repo job, don't construct local entrypoint or notebook flag
+        #    entrypoint should be already set in metadata
+        if metadata.get("_partial"):
```

Comment:
In the partial case, shouldn't we not even enter this fucntion? The job source is given by the contents of `_partial`?

Reply:
This condition is hit when the job builder attempts to build a partial job. The use_artifact `partial` stuff we download from the server only exists **after** this, when we try to construct a **full** job from a previously created partial run. Without this, the metadata is messed up on the v0 of the job, as we assume that we are running the job, and thus constructing the entrypoint from within the repo context. This is not true, as we are in the CLI context, so the metadata picks up the wrong runtime/executable. We fix this by manually fetching the git repo in `create_job` and setting the metadata from that. @KyleGoyette

Reply:
I see, so this happens when the user calls `wandb create job` the comment is somewhat unclear since the OG job could be a partial. I'd modify it to say "if building a partial job..."